### ISSUE TITLE
SDL2 (1b8ff6ac7) Movies play as static only: buffer ring overrun

### ISSUE THREAD

**Issue by andrew-strong** [dxx-rebirth/dxx-rebirth#398]
### Environment

https://github.com/dxx-rebirth/dxx-rebirth/commit/1b8ff6ac7712fd3942831aef029bfcb280845b29

#### CPU environment

Slackware Linux 64bit

### Description

After compiling d2x-rebirth against sdl2 movies only play as scratchy noise.
Music and sound effects are fine during the game itself and if the game is
run as `d2x-rebirth -nomovies` it all runs flawlessly.

Game output with movies playing is:

```
andrew@illium~/source/dxx-rebirth$ d2x-rebirth

Type 'd2x-rebirth -help' for a list of command-line options.

D2X-Rebirth v0.59.100  Jul 31 2018 16:04:48
This is a MODIFIED version of Descent 2, based on Full Version v1.2.
Copyright (C) 1994-1996 Parallax Software Corporation
DESCENT is a trademark of Interplay Productions, Inc.
Copyright (C) 1999 Peter Hawkins, 2002 Bradley Bell, 2005-2013 Christian Beckhaeuser, 2013-2017 Kp
Using SDL_mixer library v2.0.2
sdl-joystick: 0 joysticks
DXX-Rebirth: OpenGL: disabling automatic GL sync since VSync is turned off
d'oh!  buffer ring overrun (16)
You have no Smart Missiles!
Smart Mine selected!
Mega Missile selected!
Your Laser is maxed out!
Energy boosted to 158
EXTRA LIFE!
Wall opened!
You have escaped the mine!
d'oh!  buffer ring overrun (16)
d'oh!  buffer ring overrun (37)
andrew@illium~/source/dxx-rebirth$ 

```

### Engines affected:
  * [ ] D1X-Rebirth
  * [x ] D2X-Rebirth
  
I have not tested d1x-rebirth

**Comment by vLKp** [comment]
Thanks for the report.  Confirmed broken in D2X-Rebirth.  The specific code patches is only used for Descent 2 movies, so I think Descent 1 is not affected.

This is the same general cause as #396.  In both issues, the game had historically told SDL_mixer to read more data than was actually defined.  With SDL1, this was silently permitted.  With SDL2, it causes the unwanted noise.  Fixing #396 did not fix this issue, because the conversion code was duplicated in the MVE logic.  After I saw your report, I used `git grep` to search for more uses of `SDL_ConvertAudio` (the key function that Rebirth used in both places).  Only the hit for #396 and the hit for this bug were found, so I think there should not be any more instances of this specific bug lurking.

[SDL2: MVE: fix noise from reading undefined sound data](/dxx-rebirth/dxx-rebirth/commit/524f0426590dda6b55ba4aadc0fe7debb5cf6b3a) resolves this for me.
